text/0000-add-hyperlinks.md

```diff
@@ -0,0 +1,84 @@
+
+# Design
+
+This proposal is for adding a `links` property to the `meta` object in Refract elements.
+
+- `links` (array)
+    - (object)
+        - `rel` - Link relation of the link
+        - `href` - Resolvable URL for the link
+
+The `rel` property SHOULD conform to [RFC 5988](https://tools.ietf.org/html/rfc5988). Link relations should be as registered or extension link relations, and MAY be defined in profile documents.
+
+## Use Cases
+
+As mentioned, it is becoming common to need to add domain-specific information into documents. Providing hyperlinks allows to convey this kind of information without adding additional namespace specifications. The downside of additional namespaces is that they require new media types and new libraries. Baking in hyperlinks can provide an avenue that does not require new namespaces.
+
+This shows up when trying to convey state of an element or resource. For instance, the presence of an `edit` link relation in an element tells the client not only how to edit the element, but also that it can edit it. If the link is not there, the client knows it cannot edit that given element or resource.
+
+Additionally, it is common to provide properties that couple the format with certain state and functionality. In the above example, a designer may include `canEdit: true` as a property in an element's attributes. To accomplish this, the designer needs to document this somewhere, but that documentation is now out of band.
+
+## Example
+
+This example shows a `foo` element with two links. The presence of the `edit` link tells the client the user can edit the given element or document, and the `self` link provides a link to the actual resource.
+
+```js
+{
+  element: 'foo',
+  meta: {
+    links: [
+      {
+        rel: 'edit',
+        href: '/bar/1'
+      },
+      {
+        rel: 'self',
+        href: '/bar/1'
+      }
+    ]
+  }
+}
+```
+
+## Embeddeding Links
+
+In some situations, it may be advantageous to embed the resources themselves that would be found when the link was resolved. This current proposal is not addressing this yet, though it may be something we should look into in this RFC. This would happen if the design would go better together.
+
+## Additional Benefits
+
+Additionally, if hyperlinks are used in this way, we may be able to remove the concept of namespaces from the base Refract specification. The namespace properties allowed for programmatically including additional elements, but so far no library is using or has implemented this functionality. Since we've been using content types for extending Refract documents, we could use the `profile` link relation to define additional functionality and link relations.
+
+If we go this route, we can make the following changes to the base Refract specification.
+
+- Remove `prefix` and `namespace` from the `meta` object
```

Comment:
I am removing this from the current RFC discussion, but some thoughts (feel free to continue questions/thoughts here).

The current usage and implementation of Refract completely ignore everything dealing with namespacing. In adding hyperlinking, we can link to a `profile` that defines the usage for a given structure, which is what we're currently doing.

To actually make namespacing work, we will need to defined way to not only define namespaces (which we don't have nor will we have for a long while) but also a way to compare a given namespace with the actual data. Lots of work here to do before this could ever be supported.

Because there are no implementations or specifications even using this, and because this would require a great deal of work, and because this is a pre-1.0.0 version of things, we need to remove any cruft we can, and therefore we are thinking to remove namespacing.

What are your thoughts?